### ISSUE TITLE
Make helper function to export MLModel instances from Elasticsearch

### ISSUE THREAD

**Issue by sethmlarson** [elastic/eland#192]
Useful when you've trained your model on a development cluster and want to export and reimport into a production cluster.

Was thinking a `to_json()` which takes a `buf` and has an argument `compress` which gzip-compresses?

cc @Winterflower @stevedodson for additional ideas here

**Comment by stevedodson** [comment]
@sethmlarson - this is a great area to add some really useful APIs. Maybe we should think about not just model import/export but a sub-area of features around model management.

For instance, it would be really useful to:

- list all the models in a cluster
- delete all the models in a cluster (this is a pain at the moment)
- dump into an interpretable form the models in a cluster (show features, encoding schemes and even graph the tree)
- import an external model into the cluster from an Elastic library (e.g. import a DGA model)
- import/export models between clusters

@tveasey may have more as well

We will have some of this management in Kibana over time, but performing low-level admin tasks and deep detail into model structure and debug would be really useful in eland.

**Comment by tveasey** [comment]
This is a great initiative.

> tveasey may have more as well

One thing that springs to mind is that we have a lot of training information now written into the cluster: validation loss curves, hyperparameters chosen in each optimisation round, round duration, etc. We plan to expose this information in kibana dashboard, but I also think it makes sense to be available in say a notebook if that's your preferred environment.

We should also have a think about exposing quality measures for different models as well: this likely needs additional work on the ES side, but is a longer term vision we have around workflow.
